docs/conditions.md

```diff
@@ -0,0 +1,150 @@
+| :------------- |:-------------:|:-----:|:-----:|
+| OperatorAvailable | True | True | True |
+| OperatorProgressing | False | True | True |
+| OperatorDegraded | False | False | True |
+| Meaning | Component is 100% healthy and the Operator is idle | Component is functional but, either upgrading or healing | Component is functioning below capacity and an upgrade or heal is in progress |
+
+| Condition        | Status           | Status  |
+| :------------- |:-------------:|:-----:|
+| OperatorAvailable | False | False |
+| OperatorProgressing | False | True |
+| OperatorDegraded | True | True |
+| Meaning | Component and operator are in a failed state that requires human intervention.  Failed upgrade or failed heal | Component is in a failed state and an operator is healing |
+
+| Condition        | Status           |
+| :------------- |:-------------:|
+| OperatorAvailable | False |
+| OperatorProgressing | True |
+| OperatorDegraded | False |
+| Meaning | Operator is deploying the component |
+
+## Readiness Probe
+With a standardized set of conditions, the HCO should report the health of the
+overall application back to OLM and the user.  This will be critial for sensitive
+operations like upgrade, because OLM needs to know it shouldn't replace an
+operator when it is in the middle of imporant work.
+
+See this [issue](https://github.com/operator-framework/operator-lifecycle-manager/issues/922) for why we only want to report a readiness probe on the HCO
+instead of on all component operators.
+
+## Reason
+`Reason` is _a one-word CamelCase reason for the condition's last transition_.
+
+We'll use a series of lifecycle inspired prefixes paired with postfixes to
+standardize values for `Reason`.
+
+|         | -Failed  | -Succeeded | -Invalid | -InProgress |
+| :------------- |:-------------:|:-----:|:-----:|:-----:|
+| Install- | InstallFailed | InstallSucceeded | InstallInvalid | InstallInProgress |
+| Upgrade- | UpgradeFailed | UpgradeSucceded | UpgradeInvalid | UpgradeInProgress |
+| Heal- | HealFailed | HealSucceeded | HealInvalid | HealInProgress |
+| Configuration- | ConfigurationFailed | ConfigurationSucceeded | ConfigurationInvalid | ConfigurationInProgress |
+
+|         | -Failed  | -Succeeded | -Invalid | -InProgress |
+| :------------- |:-------------:|:-----:|:-----:|:-----:|
+| Meaning | The attempted operation **Failed** and the error is clear to the operator | The attempted operation **Succeeded** |  The attempted operation is missing something or is **Invalid** at this time | The attempted operation is **InProgress** |
+
+## Message
+`Message` is a _human-readable message indicating details about last transition_.
+
+Explain why your CR has `Reason`.
+
+
+## HCO conditions
+It's important to point out that the `ConditionTypes` on the HCO don't represent
+the condition the HCO is in, rather the condition of the component CRs.
+
+The HCO will use the same `ConditionType`s and `Reason`s, but it will be the
```

Comment:
Again, the scenarios in which an operator could be `Degraded` and `Unavailable` can't be pinned down to one Reason. I think we should skip Reason for now.

Reply:
Works for me. Let's drop reasons for now.